jaxrs/src/main/java/org/jboss/aerogear/unifiedpush/rest/registry/installations/InstallationRegistrationEndpoint.java

```diff
@@ -221,33 +230,52 @@ public Response registerInstallation(
     @Consumes(MediaType.APPLICATION_JSON)
     @Produces(MediaType.APPLICATION_JSON)
     @ReturnType("org.jboss.aerogear.unifiedpush.rest.EmptyJSON")
-    public Response increasePushMessageReadCounter(@PathParam("id") String pushMessageId,
-                           @Context HttpServletRequest request) throws IOException {
-
-        // find the matching variation:
-        final Variant variant = loadVariantWhenAuthorized(request);
-        if (variant == null) {
-            return create401Response(request);
-        }
-
-        if (pushMessageId != null) {
-            try (final InputStream props = this.getClass().getResourceAsStream("/kafka/producer.props")) {
-                final Properties properties = new Properties();
-                properties.load(props);
-
-                final Producer<String, String> producer = new KafkaProducer<>(properties);
-                producer.send(new ProducerRecord<String, String>("installationMetrics", pushMessageId, variant.getVariantID()));
-                producer.close();
-            }
-
-            // let's do update the analytics
-            metricsService.updateAnalytics(pushMessageId, variant.getVariantID());
-        } else {
-            return Response.status(Status.BAD_REQUEST).build();
-        }
-
-        return Response.ok(EmptyJSON.STRING).build();
-    }
+	public Response increasePushMessageReadCounter(@PathParam("id") String pushMessageId,
+			@Context HttpServletRequest request) throws IOException {
+
+		// find the matching variation:
+		final Variant variant = loadVariantWhenAuthorized(request);
+		if (variant == null) {
+			return create401Response(request);
+		}
+
+		if (pushMessageId != null) {
+
+			// start the producer and push a message to installation metrics
+			// topic
+			final Properties properties = ConfigurationUtils.loadProperties(KAFKA_PRODUCER_PROPERTIES_PATH);
+			final Producer<String, String> producer = new KafkaProducer<>(properties);
+			producer.send(new ProducerRecord<String, String>(KAFKA_INSTALLATION_TOPIC, pushMessageId,
+					variant.getVariantID()));
+			producer.close();
+
+			// start a consumer that reads from "istallationMetrics" and based
+			// on the information updates the analytics
+			// read only messages which weren't read from other consumers
+			final Properties consumerProperties = ConfigurationUtils.loadProperties(KAFKA_CONSUMER_PROPERTIES_PATH);
+			final KafkaConsumer<String, String> consumer = new KafkaConsumer<>(consumerProperties);
+			consumer.subscribe(Arrays.asList(KAFKA_INSTALLATION_TOPIC));
+
+			ConsumerRecords<String, String> records = consumer.poll(100);
+			for (ConsumerRecord<String, String> record : records) {
+				// the key of each record - pushMessageId
+				// the value - variant's id
+				// let's do update the analytics
+				logger.info(String.format("Update metric analytics for push message's ID %s and variant's ID %s",
```

Comment:
@polinankoleva : the reason is simple. If you do `String.format`, the result of `String.format` is sent to the logger, and then, inside the logger logic, it is checked whether the log is enabled or not.
If you use slf4j syntax, the params are sent as they are, then logger logic checks if the log is enabled or not and, only if it is enabled, performs the interpolation and prints the log.